src/services/UsersService.ts

```diff
@@ -82,21 +82,39 @@ export class UsersService {
   }
 
   async cancelUserIndividualSubscriptions(customerId: User['customerId']): Promise<void> {
+    if (!customerId) throw new Error('customerId required');
     const activeSubscriptions = await this.paymentService.getActiveSubscriptions(customerId);
 
     if (activeSubscriptions.length === 0) {
       throw new Error('Subscriptions not found');
     }
 
     const individualSubscriptions = activeSubscriptions.filter(
-      (subscription) => subscription.metadata.is_teams !== '1',
+      (subscription) => subscription.metadata.is_teams !== '1' && subscription.product?.metadata.type !== 'business',
     ) as Stripe.Subscription[];
 
     for (const subscriptionToCancel of individualSubscriptions) {
       await this.paymentService.cancelSubscription(subscriptionToCancel.id);
     }
   }
 
+  async cancelUserB2BSuscriptions(customerId: User['customerId']): Promise<void> {
+    if (!customerId) throw new Error('customerId required');
```

Comment:
This is the service, the type does not indicate that the customerId is null or undefined, validate it on the controller